src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -2050,6 +2078,82 @@ def _process_derived_type_decln(self, parent, decl, visibility_map):
             tsymbol.datatype = UnsupportedFortranType(str(decl))
             tsymbol.interface = UnknownInterface()
 
+    def _process_derived_type_contains_block(self, parent, decl, tsymbol):
+        '''
+        Process the supplied fparser2 parse tree for a the CONTAINS section of
+        a derived-type declaration to add any procedures to the DataTypeSymbol.
+
+        :param parent: PSyIR node in which to insert the symbols found.
+        :type parent: :py:class:`psyclone.psyir.nodes.ScopingNode`
+        :param decl: fparser2 parse tree of declaration to process.
+        :type decl: :py:class:`fparser.two.Fortran2003.Type_Declaration_Stmt`
+        :param tsymbol: the DataTypeSymbol representing the derived-type.
+        :type tsymbol: :py:class:`psyclone.psyir.symbols.DataTypeSymbol`
+
+        :returns: None
+
+        '''
+
+        contains_blocks = walk(decl, Fortran2003.Type_Bound_Procedure_Part)
+        if contains_blocks:
```

Comment:
To reduce subsequent indentation, I suggest changing this to:
```python
if not contains_blocks:
    return
```

Reply:
Indeed :)